### DISCUSSION TITLE
RPi4 HOWTO boot from /dev/sda after bootstrapping from /dev/mmcblk0

### DISCUSSION THREAD

**Discussion by RanchoHam** [siderolabs/talos#8288] [Q&A]
I get a working control plane using `talosctl --interactive ...`, but when I reboot without the sd card, it tries to load a boot image from some Raspberry Pi server.

I'm trying to build a cluster of 7 RPi4 cards each with 64GB of SSD.  Eventually I'd like to run some infrastructure tasks on the cluster in an air-gapped setting.  In the beginning I'm just trying to get the damn thing to boot off of the SSD without the sd card.  

I admit I am a total N00B to Talos, but I started in computers on an IBM System360.  I will pay my stupid taxes, but it shouldn't be this hard.

**Comment by steverfrancis** [comment]
What image did you put on the sd card?
If you copied a disk image (which is what https://www.talos.dev/v1.6/talos-guides/install/single-board-computers/rpi_generic/ shows), then Talos thinks it is already installed, and won't install itself to the SSD.

@frezbo how do you get the image on an SSD? Would you use https://etcher.balena.io/ or something to copy the talos .ISO to the SD, then boot off that and do an install from there?
Or boot some other OS on the SD, and then DD the Talos image to the SSD?

**Reply by RanchoHam** [reply]
BTW, I am using a RPi4 8GB running Raspberry Pi OS (Debian derivative) as my development machine.

First off I was following **https://www.talos.dev/v1.6/talos-guides/install/single-board-computers/rpi_generic/** and downloaded **metal-rpi_generic-arm64.raw.xz** which I decompressed with **unxz**.  I then used 
`sudo dd if=metal-rpi_generic-arm64.raw of=/dev/sdc conv=fsync bs=4M status=progress`
 to write it to an 8GB sd card for the target RPi4 8GB which has a 60GB SSD attached by USB3 that ends up at /dev/sda.  That resulted in a continuously rebooting target RPi4.

Next I followed  **https://www.talos.dev/v1.6/introduction/getting-started/** and did 
`wget https://github.com/siderolabs/talos/releases/download/v1.6.4/metal-arm64.raw.xz`, 
decompressed it, and again wrote it to an 8GB sd card for the target machine.

Still following the **Getting Started** I executed 
`talosctl gen config bramble https://192.168.103.18:6443` 
`talosctl apply-config --insecure -n 192.168.103.18 --file controlplane.yaml`
and I was able to get the control plane up and running, but it was still apparently bound to the sd card, because it would not boot from the 60GB SSD drive at /dev/sda.

Thanks for looking (on your holiday no less!),
Rich

**Reply by frezbo** [reply]
> @frezbo how do you get the image on an SSD? Would you use https://etcher.balena.io/ or something to copy the talos .ISO to the SD, then boot off that and do an install from there?

The iso would not work for pi, since it needs additional firmware blobs, so have to use the rpi_generic image

**Comment by steverfrancis** [comment]
So you just need to dd the image to the SSD on USB3, and you should be good
to go...
